src/VisualStudio/Core/Test/ProjectSystemShim/VisualStudioRuleSetTests.vb

```diff
@@ -11,6 +11,18 @@ Imports Roslyn.Test.Utilities
 
 Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.ProjectSystemShim
     Public Class VisualStudioRuleSetTests
+        Implements IDisposable
+
+        Private ReadOnly _tempPath As String
+
+        Public Sub New()
+            _tempPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName())
```

Comment:
❕ If possible without breaking the tests, replace `Path.GetTempPath()` with `TempPath.Root`. This ensures the temporary directory doesn't get littered even if cleanup fails.

Reply:
@sharwell You mean something else? my IDE can't find TempPath.Root.

Reply:
Sorry, it's [`TempRoot.Root`](https://github.com/dotnet/roslyn/blob/2a2b67d6ffb4f5a3178daa659662c41ea81023d0/src/Test/Utilities/Portable/TempFiles/TempRoot.cs#L13).